feat: support DROP MEASUREMENT in specific retention policy

StepY1aoZz opened the pull request:
<!-- Thank you for contributing to openGemini! -->

<!-- Mark the checkbox [X] or [x] if you agree with the item. -->

### What problem does this PR solve?

resolve #639 

### What is changed and how it works?
Now in the `drop measurement` command users can specify a retention policy when deleting measurements. If the retention policy is not specified in command, the default retention policy of current database will be used.
This feat is useful when users have many measurements in different retention policies but with same  measurement name.

Example:
```
// drop measurement in specified retention policy
DROP MEASUREMENT <retention_policy_name>.<measurement_name>

// drop measurement in default retention policy
DROP MEASUREMENT <measurement_name>
``` 

### How Has This Been Tested?
A new test case was added into server_test.go.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

localvar commented on lib/metaclient/meta_client.go:
```diff
@@ -187,7 +187,7 @@ type MetaClient interface {
 	MarkDatabaseDelete(name string) error
 	MarkRetentionPolicyDelete(database, name string) error
-	MarkMeasurementDelete(database, mst string) error
+	MarkMeasurementDelete(database, mst, rpName string) error
```
propose change the order of the parameters to make it align with other functions(like `Measurement` below).

```suggestion
	MarkMeasurementDelete(database, rpName, mst string) error
```

localvar replied:
propose change the order of the parameters to make it align with other functions(like `Measurement` below).

```suggestion
	MarkMeasurementDelete(database, rpName, mst string) error
```

localvar replied:
propose change the order of the parameters to make it align with other functions(like `Measurement` below).

```suggestion
	MarkMeasurementDelete(database, rpName, mst string) error
```

localvar commented on lib/metaclient/meta_client.go:
```diff
@@ -1278,16 +1278,17 @@ func (c *Client) MarkMeasurementDelete(database, measurement string) error {
 
 	var policy string
-	for _, rp := range dbi.RetentionPolicies {
+	if rpName == "" {
```
`rpName` or `policy` could be removed to simpilify the code.

xiangyu5632 commented on app/ts-store/run/server_test.go:
```diff
@@ -385,7 +385,7 @@ func (client *MockMetaClient) MarkDatabaseDelete(name string) error {
 	return nil
 }
-func (client *MockMetaClient) MarkMeasurementDelete(database, mst string) error {
+func (client *MockMetaClient) MarkMeasurementDelete(database, mst, rpName string) error {
```
instead 'rpName' with rpname may be better, or 'rp_name'

localvar replied:
`rp_name` should not be used, refer https://google.github.io/styleguide/go/decisions#underscores.
`rpName` and `rpname` is acceptable, but for consistency with `database` and `mst`, `rp` or `retentionPolicy` is better, and `policy` is already used in the code, it is also an option.

StepY1aoZz commented:
All the feedbacks are incorporated in the latest commit, including:

1. changed the order of arguments to `database,policy,measurement`
2. rename the argument `rpName` to `policy`
3. simplified the logic in func `MarkMeasurementDelete`

Thanks for these useful suggestions. @localvar @xiangyu5632
